### PR TITLE
Add visitor to handle received messages

### PR DESCRIPTION
**Pull request by nacho692**

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        final byte[] hash = message.getBlockHash();
+        this.blockProcessor.processBodyRequest(sender, message.getId(), hash);
+    }
+
+    public void apply(BodyResponseMessage message) {
+        this.syncProcessor.processBodyResponse(sender, message);
+    }
+
+    public void apply(NewBlockHashesMessage message) {
+        blockProcessor.processNewBlockHashesMessage(sender, message);
```
Comment (colltoaction) [inline rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java]: Missed the `blockProcessor.hasBetterBlockToSync` condition in this message

File: rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                recordEvent(sender, EventType.VALID_TRANSACTION);
+            }
+        }
+
+        transactionGateway.receiveTransactionsFrom(txs, sender.getPeerNodeID());
+
+        loggerMessageProcess.debug("Tx message process finished after [{}] nano.", System.nanoTime() - start);
+    }
+
+    public void apply(GetBlockHeadersMessage message) {
```
Comment (colltoaction) [inline rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java]: We should remove the `GetBlockHeadersMessage` class

File: rskj-core/src/main/java/co/rsk/net/sync/SyncState.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
     void newBlockHeaders(List<BlockHeader> chunk);
 
     // TODO(mc) don't receive a full message
-    void newBody(BodyResponseMessage message, MessageChannel peer);
```
Comment (colltoaction) [inline rskj-core/src/main/java/co/rsk/net/sync/SyncState.java]: Why did you swap these parameters?

File: rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java (hunk 1 above)
Comment (nacho692) [reply rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java]: Done!

File: rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java (hunk 2 above)
Comment (nacho692) [reply rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java]: Yup

File: rskj-core/src/main/java/co/rsk/net/sync/SyncState.java (hunk 3 above)
Comment (nacho692) [reply rskj-core/src/main/java/co/rsk/net/sync/SyncState.java]: No reason, old refactor

**Pull request merged**